src/sysext/sysext.c

```diff
@@ -39,16 +39,52 @@
 #include "user-util.h"
 #include "verbs.h"
 
-static char **arg_hierarchies = NULL; /* "/usr" + "/opt" by default */
+static char **arg_hierarchies = NULL; /* "/usr" + "/opt" by default for sysext and /etc by default for syscfg */
 static char *arg_root = NULL;
 static JsonFormatFlags arg_json_format_flags = JSON_FORMAT_OFF;
 static PagerFlags arg_pager_flags = 0;
 static bool arg_legend = true;
 static bool arg_force = false;
 
+/* Is set to true only when sysext is called with the syscfg functionality instead of the default */
+static bool syscfg_setting = false;
```

Comment:
hmm, so I think this should be prefixed with "arg_". It's not directly an argument of the tool, but indirectly it is, after all its derived from argv[0]...

Also, instead of a boolean maybe just reuse the ImageClass enum here, so that this becomes:

```c
static ImageClass arg_image_class = IMAGE_SYSEXT;
```

And then we could even replace all those little functions below that return one or the other string like this:

```c
static const struct {
        const char *directory_name;
        const char *short_identifier;
        const char *short_long_identifier;
        const char *short_identifier_plural;
        const char *level_env;
        const char *scope_env;
} image_class_info[_IMAGE_CLASS_MAX] = {
        [IMAGE_EXTENSION] = {
                .directory_name = ".systemd-syscfg",
                .short_identifier = "syscfg",
                .long_identifier = "sysext",
                …
        },
        [IMAGE_SYSCFG] = {
                .directory_name = ".systemd-sysext",
                …
};
```

you get the idea. then you can access the right values via `image_class_info[arg_image_class].directory_name` and similar.

Reply:
you can even put function pointers here, that point to the right parser/validation functions for the image class.